apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx

```diff
@@ -425,21 +550,29 @@ const ScreeningDecisionDialogApi = ({
 
   return (
     <ScreeningDecisionDialog
-      poolCandidate={{} as PoolCandidate} // TODO: Add pool candidate here
-      skill={{} as Skill} // TODO: Add skill here
-      assessmentStep={{} as AssessmentStep} // TODO: Add assessment step here
-      initialValues={{} as FormValues} // TODO: Add assessment result initial values
+      poolCandidate={poolCandidate}
+      assessmentStep={assessmentStep}
+      poolSkill={poolSkill}
+      initialValues={initialValues}
+      hasBeenAssessed={hasBeenAssessed}
+      educationRequirement={educationRequirement}
       onSubmit={(formValues) =>
-        assessmentResultId
+        hasBeenAssessed
           ? handleUpdateAssessment(
               convertFormValuesToApiUpdateInput({
                 formValues,
-              } as FormValuesToApiUpdateInputArgs),
+                assessmentResultId,
+                assessmentResultType,
+              }),
             )
           : handleCreateAssessment(
               convertFormValuesToApiCreateInput({
                 formValues,
-              } as FormValuesToApiCreateInputArgs),
+                assessmentResultType,
+                assessmentStepId: assessmentStep.id,
+                poolCandidateId: poolCandidate.id,
+                skillId: poolSkill?.id || "",
+              }),
```

Comment:
I think this prop should be renamed poolSkillId to avoid confusion as they are separate models/tables

Reply:
I agree! Changed by [7c942f7](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9324/commits/7c942f7090a919c24ca560cdc3555bdeb39eda44)